api/v1beta1/grafanalibrarypanel_types.go

```diff
@@ -0,0 +1,145 @@
+package v1beta1
+
+import (
+	"time"
+
+	"github.com/grafana/grafana-operator/v5/controllers/metrics"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GrafanaLibraryPanelSpec defines the desired state of GrafanaLibraryPanel
+// +kubebuilder:validation:XValidation:rule="(has(self.folderUID) && !(has(self.folderRef))) || (has(self.folderRef) && !(has(self.folderUID))) || !(has(self.folderRef) && (has(self.folderUID)))", message="Only one of folderUID or folderRef can be declared at the same time"
+// +kubebuilder:validation:XValidation:rule="(has(self.folder) && !(has(self.folderRef) || has(self.folderUID))) || !(has(self.folder))", message="folder field cannot be set when folderUID or folderRef is already declared"
+// +kubebuilder:validation:XValidation:rule="((!has(oldSelf.uid) && !has(self.uid)) || (has(oldSelf.uid) && has(self.uid)))", message="spec.uid is immutable"
+type GrafanaLibraryPanelSpec struct {
+	GrafanaCommonSpec  `json:",inline"`
+	GrafanaContentSpec `json:",inline"`
+
+	// folder assignment for dashboard
+	// +optional
+	FolderTitle string `json:"folder,omitempty"`
```

Comment:
Name based matching is very finicky and gets even worse with nested folders.

I'd get rid of this entirely as we have `FolderUID` and `FolderRef` available